landingai/pipeline/image_source.py

```diff
@@ -141,11 +141,36 @@ def __init__(self, uri: str, samples_per_second: float = 1) -> None:
         uri : str
             URI to the video file. This could be a local file or a URL that serves the video file in bytes.
         samples_per_second : float, optional
-            The number of images to sample per second (by default 1)
+            The number of images to sample per second (by default 1). If set to zero, it disables sampling
         """
         self._video_file = str(fetch_from_uri(uri))
         self._local_cache_dir = Path(tempfile.mkdtemp())
         self._samples_per_second = samples_per_second
+        cap = cv2.VideoCapture(self._video_file)
+        self._src_fps = cap.get(cv2.CAP_PROP_FPS)
+        self._src_total_frames = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+        cap.release()
+        if self._samples_per_second != 0:
+            # Adjust the video frames and fps based on sampling rate
+            self._target_total_frames = int(
+                self._src_total_frames * self._samples_per_second / self._src_fps
+            )
+            self._target_fps = self._samples_per_second
+        else:
+            self._target_total_frames = self._src_total_frames
+            self._target_fps = self._src_fps
```

Comment:
could we add a unit test for it?

Reply:
what's better is abstract out the `fps` resolution code as a dedicated function? so you don't have to test it twice (avoid duplication in `landingai/image_source_ops.py`)